saleor/graphql/channel/mutations.py

```diff
@@ -51,3 +54,63 @@ class Meta:
         permissions = (ChannelPermissions.MANAGE_CHANNELS,)
         error_type_class = ChannelError
         error_type_field = "channel_errors"
+
+
+class ChannelDeleteInput(graphene.InputObjectType):
+    target_channel = graphene.ID(
+        required=True,
+        description="ID of channel to migrate orders from origin channel.",
+    )
+
+
+class ChannelDelete(ModelDeleteMutation):
+    class Arguments:
+        id = graphene.ID(required=True, description="ID of a channel to delete.")
+        input = ChannelDeleteInput(
+            required=True, description="Fields required to delete a channel."
+        )
+
+    class Meta:
+        description = "Delete a channel."
```

Comment:
This description is part of the documentation so we can use it to describe the behavior a bit:

"Delete a channel. Orders associated with the deleted channel will be moved to the target channel. Checkouts, product availability, and pricing will be removed."